uint16 total
float64[30] x
float64[30] y
float64[30] r